drivers/density/simulator.py

```diff
@@ -62,9 +61,13 @@ def run_gate_array(gate_array):
         else:
             expand_single_gate(simulator_gate, gate.i, num_qbits) | qubits
 
-    print "Final distribution:", qubits.distribution()
-    measurement = [measure(qubits, i) for i in xrange(num_qbits)]
-    print "Qubit measure:", ", ".join((str(q) for q in measurement))
+    if print_dist:
+        print "Final distribution:", qubits.distribution()
+    if print_state:
+        print "Final state:", qubits.state
+    for _ in xrange(num_measures):
+        measurement = [measure(qubits, i, project=False) for i in xrange(num_qbits)]
+        print "Qubit measure:", ", ".join((str(q) for q in measurement))
```

Comment:
Maybe print the probabilities for 0,1 from those measurements as well? Or we could print the probability for 0/1 from the sum of all statens where it's either

Reply:
Don't understand exactly. But I think we should add more complicated things in the qcode language itself with the goal of removing this last measurement step entirely so it is explicit in the source code what you want to do instead.

https://github.com/odanielson/QuantumProgramming/pull/26 is a first step in this direction.